cmake/compile_definitions/windows.cmake

```
@@ -30,6 +30,14 @@ file(GLOB NVPREFS_FILES CONFIGURE_DEPENDS
         "${CMAKE_SOURCE_DIR}/src/platform/windows/nvprefs/*.cpp"
         "${CMAKE_SOURCE_DIR}/src/platform/windows/nvprefs/*.h")
 
+include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/third-party/nvenc-headers")
+file(GLOB_RECURSE NVENC_SOURCES CONFIGURE_DEPENDS
+        "${CMAKE_SOURCE_DIR}/src/nvenc/*.h"
+        "${CMAKE_SOURCE_DIR}/src/nvenc/*.cpp")
+
+include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/third-party/nvenc-headers/12.0/include")
```

```suggestion
include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/third-party/nvenc-headers/1200/include")
```

Unless I'm missing something, this path doesn't exist.

---

Yeah, it used what we ship with ffmpeg in this case. I will think about how to handle this best, either remove these particular headers from windows ffmpeg, or somehow use them (not trivial and may have drawbacks).

---

Decided to drop this "default version" logic to minimize potential problems.
One extra version of some classes will be compiled with ffmpeg version of nvenc sdk, but then discarded by the linker.